newsfeeds/api/views.py

```diff
@@ -20,8 +18,11 @@ def get_queryset(self):
         return NewsFeed.objects.filter(user=self.request.user)
 
     def list(self, request):
-        newsfeeds = NewsFeedService.get_cached_newsfeeds(request.user.id)
-        page = self.paginate_queryset(newsfeeds)
+        cached_newsfeeds = NewsFeedService.get_cached_newsfeeds(request.user.id)
+        page = self.paginator.paginate_cached_list(cached_newsfeeds, request)
+        if page is None:
```

Comment:
if page is none, it means there is no related data in the cache. we need to get them from the db